### PR TITLE
AT-3276: shortened http timeout & refactoring

### PR DESCRIPTION
**Pull request by 23tux**

### CODE REVIEW

File: spec/atalogics_api/v3/client_spec.rb [hunk 1]
```diff
@@ -26,45 +28,45 @@
...
   end
 
-  it 'expires a key, when cached catch_window is older than Time.now' do
+  it "expires a key, when cached catch_window is older than Time.now" do
     # current time is one second after catch_window usable_till
     Timecop.freeze("2016-02-02T17:00:01")
 
     cache_key = "V3_/offers_bar"
-    cached_body = {"offers" => [{"catch_window" => {"usable_till" => "2016-02-02T17:00:00+01:00"}}]}
+    cached_body = { "offers" => [{ "catch_window" => { "usable_till" => "2016-02-02T17:00:00+01:00" } }] }
```
Comment (enthusiasmus) [inline spec/atalogics_api/v3/client_spec.rb:39]: In contrast to line 36, line 39 has zone information

File: spec/atalogics_api/v3/client_spec.rb [hunk 2]
```diff
@@ -26,45 +28,45 @@
...
     AtalogicsApi.cache_store.set(cache_key, [200, cached_body].to_json)
 
-    expect(client.class).to receive(:post).and_return(double("httparty_response", code: 200, parsed_response: {new: "response"}))
+    expect(client.class).to receive(:post).and_return(double("httparty_response", code: 200, parsed_response: { new: "response" }))
```
Comment (enthusiasmus) [inline spec/atalogics_api/v3/client_spec.rb:42]: We should use here `instance_double`

File: spec/atalogics_api_spec.rb [hunk 3]
```diff
@@ -4,4 +4,10 @@
   it 'has a version number' do
     expect(AtalogicsApi::VERSION).not_to be nil
   end
+
+  describe described_class::Errors do
+    it "returns all errors" do
+      expect(described_class::ALL.length).to eq(4)
```
Comment (enthusiasmus) [inline spec/atalogics_api_spec.rb:10]: The content is actually more interesting here?

File: spec/atalogics_api/v3/client_spec.rb [hunk 4]
```diff
@@ -26,45 +28,45 @@
 
     cache_key = "V3_/offers_Auerspergstr 44 Salzburg_Getreidegasse 24 Salzburg"
     expect(AtalogicsApi.cache_store.keys).to eq([cache_key])
-    expect(AtalogicsApi.cache_store.ttl(cache_key)).to eq(24*60*60)
+    expect(AtalogicsApi.cache_store.ttl(cache_key)).to eq(24 * 60 * 60)
```
Comment (enthusiasmus) [inline spec/atalogics_api/v3/client_spec.rb:31]: You could use the constant here

File: lib/atalogics_api/v3/client.rb [hunk 5]
```diff
@@ -34,26 +41,19 @@ def offers body
       # CACHEABLE
       # @param body [Hash] Hash with address parts, see docs
       # @return [HTTParty::Response]
-      def offers! body
+      def offers!(body)
         response = offers body
-        raise Response::FailedError.new(response) if response.code != 200
+        raise Response::FailedError, response if response.code != 200
```
Comment (enthusiasmus) [inline lib/atalogics_api/v3/client.rb:46]: Subjectively speaking, here parentheses would be beautiful.
Applies to all method calls with (multiple) parameters and a conditional

File: lib/atalogics_api/client_base.rb [hunk 6]
```diff
@@ -52,65 +57,69 @@ def access_token
...
-      store_cached_result(cache_key, response.code, response.body) if cache_key
+      expires_in = options.fetch(:expires_in, EXPIRES_IN_24_HOURS)
+      store_cached_result(cache_key, response.code, response.body, expires_in) if cache_key
       response
     end
 
-    private def perform_api_request method, *args, &block
-      response = self.class.send(method, *args)
+    private def perform_http_request(method, url, options)
+      response = self.class.send(method, url, options)
```
Comment (enthusiasmus) [inline lib/atalogics_api/client_base.rb:84]: this line calls for example `get`, which calls `perform`, which calls `perform_http_request`, which calls `get`, etc.? :-D

File: .rubocop.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+Layout/DotPosition:
+  EnforcedStyle: trailing
+
+Gemspec/OrderedDependencies:
+  Enabled: false
+
+# There is a bug in rubocop 0.52.1 (current version 2018-01-12),
+# where time and date formats like "%d.%m%" are offenses
+# see https://github.com/bbatsov/rubocop/issues/5223
```
Comment (enthusiasmus) [inline .rubocop.yml:27]: This issue is already closed - so we can remove the option and the comment?

File: lib/atalogics_api/client_base.rb (hunk 6 above)
Comment (23tux) [reply lib/atalogics_api/client_base.rb:84]: nope, it calls `self.class.get` not `self.get`

File: lib/atalogics_api/v3/client.rb (hunk 5 above)
Comment (23tux) [reply lib/atalogics_api/v3/client.rb:46]: Was Rubocop auto fix

File: spec/atalogics_api/v3/client_spec.rb (hunk 1 above)
Comment (23tux) [reply spec/atalogics_api/v3/client_spec.rb:39]: when you use `Timecop.freeze("2016-02-02T17:00:01")` it knows that we are in +1 time zone

**Pull request merged**